storage/series.go

```diff
@@ -332,21 +332,27 @@ func (s *dbSeries) Bootstrap(rs m3db.DatabaseSeriesBlocks, cutover time.Time) er
 	// data accumulated during bootstrapping.
 	s.buffer.DrainAndReset(true)
 
+	// NB(xichen): if an error occurred during series bootstrap, we close
+	// the database series blocks and mark the series bootstrapped regardless
+	// in the hope that the other replicas will provide data for this series.
+	multiErr := xerrors.NewMultiError()
 	for i := range s.pendingBootstrap {
 		stream := s.pendingBootstrap[i].encoder.Stream()
 		err := s.drainStream(rs, stream, cutover)
 		stream.Close()
 		if err != nil {
-			s.Unlock()
-			return err
+			rs.Close()
+			rs = NewDatabaseSeriesBlocks(s.opts)
+			err = xerrors.NewRenamedError(err, fmt.Errorf("error occurred bootstrapping series %s: %v", s.seriesID, err))
+			multiErr = multiErr.Add(err)
 		}
 	}
 
 	s.blocks = rs
 	s.bs = bootstrapped
 	s.Unlock()
 
-	return nil
+	return multiErr.FinalError()
```

Comment:
Could you also nil out `s.pendingBootstrap` at the end here to save a tiny bit more memory? Hehe.